src/Swarm/Game/State.hs

```diff
@@ -609,6 +609,7 @@ scenarioToGameState scenario userSeed toRun g = do
       , _waitingRobots = M.empty
       , _gensym = initGensym
       , _randGen = mkStdGen seed
+      , _entityMap = g ^. entityMap <> scenario ^. scenarioEntities
```

Comment:
I agree that #509 is a bug and this is the change needed to fix it. However, we now have another problem which is what my comment `do we need to keep an old entity map around???` was talking about: changes to the entity map are permanent.  For example, to reproduce:
1. Start Swarm in cheat mode
2. Load the new testing scenario containing `Tardis`
3. Quit to the menu
4. Start a creative mode game
5. Execute `create "Tardis"`: it *should* say "I've never heard of a Tardis", but instead, it works.  That's because the `Tardis` was added to the `entityMap` when loading the test scenario, and it was never removed.

We will have the same problem with recipes.

The problem is that `scenarioToGameState` *modifies* the existing `GameState` instead of creating a brand new one from scratch.  I am actually not sure right now whether there are good reasons for this.  But I think the "right" solution, assuming it is not too difficult for some reason I can't think of right now, is to create a new `GameState` from scratch every time we load a new scenario.  But if we don't want to actually reread the entities and recipes from disk every time, we will have to save copies of the "pristine" versions somewhere, which is what my comment `do we need to keep an old entity map around???` was supposed to mean.

I don't care whether we try to fix the problem in this PR, or split out another issue and fix it later.